tests/components/unifi/test_device_tracker.py

```diff
@@ -150,31 +244,59 @@ async def test_tracked_devices(hass):
     assert client_5 is not None
     assert client_5.state == "not_home"
 
-    device_1 = hass.states.get("device_tracker.device_1")
-    assert device_1 is not None
-    assert device_1.state == "not_home"
-
     # State change signalling works
     client_1_copy = copy(CLIENT_1)
     client_1_copy["last_seen"] = dt_util.as_timestamp(dt_util.utcnow())
-    event = {"meta": {"message": "sta:sync"}, "data": [client_1_copy]}
+    event = {"meta": {"message": MESSAGE_CLIENT}, "data": [client_1_copy]}
     controller.api.message_handler(event)
-    device_1_copy = copy(DEVICE_1)
-    device_1_copy["last_seen"] = dt_util.as_timestamp(dt_util.utcnow())
-    event = {"meta": {"message": "device:sync"}, "data": [device_1_copy]}
-    controller.api.message_handler(event)
-    await hass.async_block_till_done()
 
     client_1 = hass.states.get("device_tracker.client_1")
     assert client_1.state == "home"
 
+
+async def test_tracked_devices(hass):
+    """Test the update_items function with some devices."""
+    controller = await setup_unifi_integration(
+        hass, devices_response=[DEVICE_1, DEVICE_2],
+    )
+    assert len(hass.states.async_entity_ids(TRACKER_DOMAIN)) == 2
+
+    device_1 = hass.states.get("device_tracker.device_1")
+    assert device_1
+    assert device_1.state == "home"
+
+    device_2 = hass.states.get("device_tracker.device_2")
+    assert device_2
+    assert device_2.state == "not_home"
+
+    # State change signalling work
+    device_1_copy = copy(DEVICE_1)
+    device_1_copy["next_interval"] = 20
+    event = {"meta": {"message": MESSAGE_DEVICE}, "data": [device_1_copy]}
+    controller.api.message_handler(event)
+    device_2_copy = copy(DEVICE_2)
+    device_2_copy["next_interval"] = 50
+    event = {"meta": {"message": MESSAGE_DEVICE}, "data": [device_2_copy]}
+    controller.api.message_handler(event)
+    await hass.async_block_till_done()
+
     device_1 = hass.states.get("device_tracker.device_1")
     assert device_1.state == "home"
+    device_2 = hass.states.get("device_tracker.device_2")
+    assert device_2.state == "home"
+
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=40))
+    await hass.async_block_till_done()
+
+    device_1 = hass.states.get("device_tracker.device_1")
+    assert device_1.state == "not_home"
+    device_2 = hass.states.get("device_tracker.device_2")
+    assert device_2.state == "home"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)